sklearn/inspection/_plot/partial_dependence.py

```diff
@@ -973,22 +1032,39 @@ def plot(self, *, ax=None, n_cols=3, line_kw=None, contour_kw=None):
         self.deciles_vlines_ = np.empty_like(self.axes_, dtype=object)
         self.deciles_hlines_ = np.empty_like(self.axes_, dtype=object)
 
-        for pd_plot_idx, (axi, feature_idx, pd_result) in enumerate(
-            zip(self.axes_.ravel(), self.features, self.pd_results)
+        for pd_plot_idx, (axi, feature_idx, pd_result, kind_plot) in enumerate(
+            zip(self.axes_.ravel(), self.features, self.pd_results, kind)
         ):
             avg_preds = None
             preds = None
             feature_values = pd_result["values"]
-            if self.kind == "individual":
+            if kind_plot == "individual":
                 preds = pd_result.individual
-            elif self.kind == "average":
+            elif kind_plot == "average":
                 avg_preds = pd_result.average
-            else:  # kind='both'
+            else:  # kind_plot == 'both'
                 avg_preds = pd_result.average
                 preds = pd_result.individual
```

Comment:
If `kind` is not `'both'`, this will be executed. Should a validation be done on the elements of `kind_`?